stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/listener/RegisterListener.java

```diff
@@ -76,11 +85,39 @@ public void execute(Context context) {
         Map<String, Value> queueConfig = value.getMapField(RabbitMQConstants.QUEUE_CONFIG);
         String queueName = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_NAME).getStringValue();
         boolean durable = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_DURABLE).getBooleanValue();
-        boolean exclusive = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_EXCLUSIVE).getBooleanValue();;
-        boolean autoDelete = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_AUTODELETE).getBooleanValue();;
+        boolean exclusive = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_EXCLUSIVE).getBooleanValue();
+        boolean autoDelete = queueConfig.get(RabbitMQConstants.ALIAS_QUEUE_AUTODELETE).getBooleanValue();
         ChannelUtils.queueDeclare(channel, queueName, durable, exclusive, autoDelete);
         onMessageResource = service.getResources()[0];
-        getMessages(onMessageResource, channel, queueName);
+        List<ParamDetail> paramDetails = onMessageResource.getParamDetails();
+        BType dataType = paramDetails.get(0).getVarType();
+        int dataTypeTag = dataType.getTag();
+        switch (dataTypeTag) {
+            case TypeTags.STRING_TAG:
+                dispatchTextMessage(onMessageResource, channel, queueName);
+                break;
+            case TypeTags.FLOAT_TAG:
+                dispatchFloatMessage(onMessageResource, channel, queueName);
+                break;
+            case TypeTags.ARRAY_TAG:
+                if (((BArrayType) dataType).getElementType().getTag() == TypeTags.BYTE_TAG) {
+                    dispatchBytesMessage(onMessageResource, channel, queueName);
+                } else {
+                    throw new BallerinaException("Unsupported data type for array parameter");
+                }
+                break;
+            case TypeTags.INT_TAG:
+                dispatchIntMessage(onMessageResource, channel, queueName);
```

Comment:
Shall we handle the possible exceptions in these methods? Can't we return ballerina errors?